app/Services/Manager.php

```diff
@@ -226,30 +253,55 @@ public function getUserActivity($id, $model)
         }
 
         if ($signup && $signup->reportback) {
-            // Provide the admin URL to the reportback.
-            $signup->reportback->admin_url = env('PHOENIX_URI') . '/admin/reportback/' . $signup->reportback->id;
-
-            // Format the update timestamp
-            $signup->reportback->updated_at = new Carbon($signup->reportback->updated_at);
-            $signup->reportback->updated_at = $signup->reportback->updated_at->format('Y-m-d');
-
-            // Set flagged status to 'pending' if it is NULL, otherwise use bool value
-            if (! isset($signup->reportback->flagged)) {
-                $signup->reportback->flagged = 'pending';
-            } elseif ($signup->reportback->flagged) {
-                $signup->reportback->flagged = 'flagged';
-            } else {
-                $signup->reportback->flagged = 'approved';
-            }
-
-            // Return the reportback.
-            return $signup->reportback;
+            return $this->formatReportback($signup);
         }
 
         // If the user has no activity for this competition or waiting room.
         return null;
     }
 
+    public function getUsersActivity($ids, $model)
+    {
+        $campaign = $model->contest->campaign_id;
+        $campaign_run = $model->contest->campaign_run_id;
+
+        $signups = [];
+        $count = intval(ceil(count($ids) / 50));
+        $index = 0;
+
+        for ($i = 0; $i < $count; $i++) {
+            $batch = array_slice($ids, $index, 50);
+            $signups = array_merge($signups, $this->getUserSignup(implode(',', $batch), $campaign, $campaign_run, true));
+            $index += 50;
+        }
+
+        return collect($signups)->keyBy(function ($item) {
+            return $item->user->id;
+        });
+    }
+
+    private function formatReportback($signup)
+    {
+        // Provide the admin URL to the reportback.
+        $signup->reportback->admin_url = env('PHOENIX_URI') . '/admin/reportback/' . $signup->reportback->id;
+
+        // Format the update timestamp
+        $signup->reportback->updated_at = new Carbon($signup->reportback->updated_at);
+        $signup->reportback->updated_at = $signup->reportback->updated_at->format('Y-m-d');
+
+        // Set flagged status to 'pending' if it is NULL, otherwise use bool value
+        if (! isset($signup->reportback->flagged)) {
+            $signup->reportback->flagged = 'pending';
+        } elseif ($signup->reportback->flagged) {
+            $signup->reportback->flagged = 'flagged';
+        } else {
+            $signup->reportback->flagged = 'approved';
+        }
+
+        // Return the reportback.
+        return $signup->reportback;
```

Comment:
do you want to return just the reportback or the full signup? Since you pass in the full signup, to format stuff, I would assume you'd be sending the full signup back...?